src/plugins/controls/public/control_group/component/control_group_component.tsx

```diff
@@ -254,16 +283,48 @@ export const ControlGroup = () => {
                 </DragOverlay>
               </DndContext>
             </EuiFlexItem>
-            {showAddButton && (
-              <EuiFlexItem grow={false}>
-                <EuiButtonIcon
-                  size="s"
-                  iconSize="m"
-                  display="base"
-                  iconType={'plusInCircle'}
-                  aria-label={ControlGroupStrings.management.getAddControlTitle()}
-                  onClick={() => controlGroup.openAddDataControlFlyout()}
-                />
+            {showAppendedButtonGroup && (
+              <EuiFlexItem
+                grow={false}
+                className="controlGroup--endButtonGroup"
+                data-test-subj="controlGroup--endButtonGroup"
+              >
+                <EuiFlexGroup responsive={false} gutterSize="s" alignItems="center">
+                  {showAddButton && (
+                    <EuiFlexItem grow={false}>
+                      <EuiToolTip content={ControlGroupStrings.management.getAddControlTitle()}>
+                        <EuiButtonIcon
+                          size="m"
+                          iconSize="m"
+                          display="base"
+                          iconType={'plusInCircle'}
+                          data-test-subj="controlGroup--addControlButton"
+                          aria-label={ControlGroupStrings.management.getAddControlTitle()}
+                          onClick={() => controlGroup.openAddDataControlFlyout()}
+                        />
+                      </EuiToolTip>
+                    </EuiFlexItem>
+                  )}
+                  {showApplySelections && (
+                    <EuiFlexItem grow={false}>
+                      <EuiButtonIcon
+                        size="m"
+                        disabled={!applyButtonEnabled}
+                        iconSize="m"
+                        display="fill"
+                        color={'success'}
+                        iconType={'check'}
+                        data-test-subj="controlGroup--applyFiltersButton"
+                        aria-label={ControlGroupStrings.management.getApplyButtonTitle(
+                          applyButtonEnabled
+                        )}
+                        onClick={() => {
+                          if (unpublishedFilters) controlGroup.publishFilters(unpublishedFilters);
+                        }}
+                      />
+                    </EuiFlexItem>
```

Comment:
I used to have a tooltip around the apply button that would change its text depending on if the button was enabled or not. However, there is a [known bug](https://github.com/elastic/eui/issues/6488) with React where the tooltip would get completely stuck when the button changed from enabled to disabled:

https://github.com/elastic/kibana/assets/8698078/d31e4110-e0ff-45a8-a046-4262b6486985

As a result of this, I've decided to remove the tooltip for now. In theory, we could have a tooltip in **either** the enabled or disabled state, but **not** both; I wasn't sure which one should take priority, though, and it felt a bit strange in this case to only have a tooltip in one state....

Reply:
@andreadelrio

Reply:
I tested the solution locally using Safari + VO. The button drives really well after it's enabled and will be a nice way to put the power to update immediately in the user's hands. If we get future pushback that a submit button is needed, we can revisit adding instructions to check the Controls menu.

I think we could add a `aria-live` region in a future ticket to let users know when their filters were added. Thinking out loud, we could populate a live region when the button is re-rendered in gray and the accessible label updated. This would be a nice way to inform screen reader users that their filters have been applied without requiring any input.

LMK if you'd like to discuss or have me put a ticket in.

Reply:
@1Copenut A separate ticket for that would be greatly appreciated 🙇 That way, we can keep discussion contained there since this PR is already... pretty large 😆

Reply:
@nickpeihl That is very fair. Interested in what @andreadelrio thinks in this case... Do we want to **hide** the apply button rather than disabling it when there are no unapplied changes?

Reply:
> I used to have a tooltip around the apply button that would change its text depending on if the button was enabled or not. However, there is a [known bug](https://github.com/elastic/eui/issues/6488) with React where the tooltip would get completely stuck when the button changed from enabled to disabled:
> 
>  Screen.Recording.2024-03-05.at.9.11.22.AM.mov 
> As a result of this, I've decided to remove the tooltip for now. In theory, we could have a tooltip in **either** the enabled or disabled state, but **not** both; I wasn't sure which one should take priority, though, and it felt a bit strange in this case to only have a tooltip in one state....

@Heenawter In my view, a tooltip is more needed in the disabled state because a disabled button is something that can lead to user confusion. I think the enabled state is pretty self-explanatory in this case and the tooltip would be a nice-to-have.

Reply:
Added in cf9cb525ed41dc520b45c471341932bdfb410020 :+1:



https://github.com/elastic/kibana/assets/8698078/0ee26454-8780-4e48-9910-118c0b4649af